src/app/clusters/camera-av-settings-user-level-management-server/camera-av-settings-user-level-management-server.cpp

```diff
@@ -0,0 +1,1079 @@
+    }
+    else
+    {
+        *it = aMptzPresetHelper;
+    }
+
+    // Update the current preset ID to the next available
+    UpdatePresetID();
+    MarkDirty(Attributes::MPTZPresets::Id);
+
+    ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Protocols::InteractionModel::Status::Success);
+}
+
+void CameraAvSettingsUserLevelMgmtServer::HandleMPTZRemovePreset(HandlerContext & ctx,
+                                                                 const Commands::MPTZRemovePreset::DecodableType & commandData)
+{
+    uint8_t presetToRemove = commandData.presetID;
+
+    // Verify the provided presetID is within spec limits
+    //
+    if (presetToRemove > mMaxPresets - 1)
+    {
+        ChipLogError(Zcl, "Preset to remove is out of range");
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::ConstraintError);
+        return;
+    }
+
+    // Is the provided ID known to us?
+    //
+    auto it = std::find_if(mMptzPresetHelpers.begin(), mMptzPresetHelpers.end(),
+                           [presetToRemove](const MPTZPresetHelper & mptzph) { return mptzph.GetPresetID() == presetToRemove; });
+
+    if (it == mMptzPresetHelpers.end())
+    {
+        ChipLogError(Zcl, "No matching presets, RemovePreset not possible");
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::NotFound);
+        return;
+    }
+
+    // Call the delegate to ensure that it is ok to remove the preset indicated.
+    //
+    Status status = mDelegate->MPTZRemovePreset(presetToRemove);
+
+    if (status != Status::Success)
+    {
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, status);
+        return;
+    }
+
+    // Remove the identified item from the known set of presets
+    //
+    mMptzPresetHelpers.erase(it);
+    MarkDirty(Attributes::MPTZPresets::Id);
+
+    ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Protocols::InteractionModel::Status::Success);
+}
+
+void CameraAvSettingsUserLevelMgmtServer::HandleDPTZSetViewport(HandlerContext & ctx,
+                                                                const Commands::DPTZSetViewport::DecodableType & commandData)
+{
+    uint16_t videoStreamID                 = commandData.videoStreamID;
+    Structs::ViewportStruct::Type viewport = commandData.viewport;
+
+    // Is this a video stream ID of which we have already been informed?
+    // If not, ask the delegate if it's ok.  If yes, add to our set and proceed, if not, fail.
+    //
+    if (!KnownVideoStreamID(videoStreamID))
+    {
+        // Call the delegate to validate that the videoStreamID is known; if yes then add to our list and proceed
+        //
+        if (!mDelegate->IsValidVideoStreamID(videoStreamID))
+        {
+            ChipLogError(Zcl, "Unknown Video Stream ID provided.");
+            ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::NotFound);
+            return;
+        }
+        AddMoveCapableVideoStreamID(videoStreamID);
+        MarkDirty(Attributes::DPTZRelativeMove::Id);
```

Comment:
AddMoveCapableVideoStreamID did that already, right?

Reply:
It did ... deleting!